src/shared_modules/content_manager/src/components/offlineDownloader.hpp

```diff
@@ -80,49 +83,98 @@ class OfflineDownloader final : public AbstractHandler<std::shared_ptr<UpdaterCo
     };
 
     /**
-     * @brief Downloads the requested local file and update the context accordingly.
+     * @brief Copy a file from the localsystem.
      *
-     * @note Despite the method name, there is no such download since the file is present in the filesystem.
+     * @param inputFilepath Input path from where to copy the file.
+     * @param outputFilepath Output path where to paste the file.
+     * @return true if the file was copied, otherwise false.
+     */
+    bool copyFile(const std::filesystem::path& inputFilepath, const std::filesystem::path& outputFilepath) const
+    {
+        constexpr auto FILE_PREFIX {"file://"};
+
+        // Remove file prefix.
+        auto unprefixedUrl {inputFilepath.string()};
+        Utils::replaceFirst(unprefixedUrl, FILE_PREFIX, "");
+
+        // Check input file existence.
+        if (!std::filesystem::exists(unprefixedUrl))
+        {
+            std::cout << "File " << unprefixedUrl << " doesn't exist. Skipping download." << std::endl;
+            return false;
+        }
+
+        // Copy file, overriding the output one if necessary.
+        std::filesystem::copy(unprefixedUrl, outputFilepath, std::filesystem::copy_options::overwrite_existing);
+        return true;
+    }
+
+    /**
+     * @brief Download a file from an HTTP server.
+     *
+     * @param inputFileURL URL from where to download the file.
+     * @param outputFilepath Output path where to store the downloaded file.
+     */
+    void downloadFile(const std::filesystem::path& inputFileURL, const std::filesystem::path& outputFilepath) const
+    {
+        const auto onError {[](const std::string& errorMessage, const long errorCode)
+                            {
+                                throw std::runtime_error {"(" + std::to_string(errorCode) + ") " + errorMessage};
+                            }};
+
+        // Download file from URL.
+        m_urlRequest.download(HttpURL(inputFileURL), outputFilepath, onError);
+    }
+
+    /**
+     * @brief Downloads a file in offline mode and updates the context accordingly.
```

Comment:
Question: Shouldn't this express that the file could be obtained from the local filesystem or from an HTTP/HTTPS server?

Reply:
Same as [above](https://github.com/wazuh/wazuh/pull/20416#discussion_r1405790888).